src/thunks.js

```diff
@@ -320,11 +322,51 @@ export function fetchAllInvoices() {
     }
     const url = '/api/invoices/';
     return Request
-            .get(url)
-            .set('Authorization', 'Bearer ' + authToken)
-            .then((response) => {
-              const normalizedInvoices = normalizeArray(response.body);
-              dispatch(receiveAllInvoices(normalizedInvoices));
-            })
+      .get(url)
+      .set('Authorization', 'Bearer ' + authToken)
+      .then((response) => {
+        const normalizedInvoices = normalizeArray(response.body);
+        dispatch(receiveInvoices(normalizedInvoices));
+      })
+  }
+}
+
+export function getInvoice(id) {
+  return (dispatch) => {
+    const authToken = getAuthToken();
+    if (!authToken) {
+      return;
```

Comment:
We should dispatch a `getInvoiceFailed` action of some kind and then return, rather than returning silently.